File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> count;

        // Count how many times each number appears
        for (int num : nums) {
            count[num]++;
        }

        // freq[i] stores all numbers that appear i times
        vector<vector<int>> freq(nums.size() + 1);

        for (auto pair : count) {
            int num = pair.first;
            int amount = pair.second;

            freq[amount].push_back(num);
        }

        vector<int> result;

        // Start from highest frequency
        for (int i = freq.size() - 1; i >= 1; i--) {
            for (int num : freq[i]) {
                result.push_back(num);

                if (result.size() == k) {
                    return result;
                }
            }
        }

        return result;
    }
};

File: Data Structures & Algorithms/is-anagram/submission-3.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        count = {}

        for letter in s:
            count[letter] = count.get(letter, 0) + 1
        for letter in t:
            count[letter] = count.get(letter, 0) - 1
        for letter in count:
            if count[letter] != 0:
                return False
        return True